osu.Game/Extensions/DrawableExtensions.cs

```diff
@@ -79,5 +80,40 @@ public static void ApplySkinnableInfo(this Drawable component, SkinnableInfo inf
                     container.Add(child.CreateInstance());
             }
         }
+
+        /// <summary>
+        /// Keeps the drawable upright and unstretched preventing it from being rotated, sheared, scaled or flipped with its Parent.
+        /// </summary>
+        /// <param name="drawable">The drawable.</param>
+        public static void KeepUprightAndUnscaled(this Drawable drawable)
+        {
+            // Decomposes the inverse of the parent FrawInfo.Matrix into rotation, shear and scale.
+            var parentMatrix = drawable.Parent.DrawInfo.Matrix;
+            parentMatrix.Transpose();
```

Comment:
What does the transposition do here?

Reply:
this transposition is not necessary (i only used it to make `parrentMatrix` a row-major matrix, because i find row-major matrices more intuitive to work with), i am removing it.